scipy/interpolate/tests/test_interpolate.py

```diff
@@ -2323,19 +2416,222 @@ def test_complex(self):
                                               method=method)
 
             v1 = interp(sample)
-            v2 = rinterp(sample) + 1j*iinterp(sample)
+            v2 = rinterp(sample) + 1j * iinterp(sample)
             assert_allclose(v1, v2)
 
+    def test_complex_exception_spline(self):
+        points, values = self._get_sample_4d()
+        values = values - 2j * values
+        sample = np.asarray([[0.1, 0.1, 1., .9]])
+
+        # spline methods dont support complex values
+        for method in self.spline_methods:
+            assert_raises(ValueError, RegularGridInterpolator, points, values,
+                          method)
+
+            interp = RegularGridInterpolator(points, values)
+            assert_raises(ValueError, interp, sample, method)
+
+    def test_minimum_required_gridsize(self):
+        for method in self.spline_methods:
+            k = self.interp_configs[method]
+            x = np.linspace(0, 1, k)
+            y = np.linspace(0, 1, k)
+            points = [x, y]
+            X, Y = np.meshgrid(*points, indexing='ij')
+            values = X + Y
+            assert_raises(ValueError, RegularGridInterpolator, points, values,
+                          method)
+
+    def test_no_gradients(self):
+        # linear and nearest interpolation don't support gradients
+        points, values = self._get_sample_4d()
+        sample = np.asarray([0.1, 0.1, 1., .9])
+        for method in ('linear', 'nearest'):
+            interp = RegularGridInterpolator(points, values)
+            assert_raises(ValueError, interp.gradient, sample, method)
+
+    def test_method_switching(self):
+        # should be able to switch interpolation methods on each __call__
+        # and gradient call, without overriding defaults permenantly.
+        # exceptions and gradient caching should work as expected.
+
+        np.random.seed(314)
+        x = np.linspace(-100, 2, 10)
+        y = np.linspace(-10, 4, 6)
+        z = np.linspace(1000, 2000, 50)
+
+        points = [x, y, z]
+        values = np.random.randn(10, 6, 50)
+
+        x = [0.5, 0, 1001]
+
+        # create as cubic
+        interp = RegularGridInterpolator(
+            points, values, method='cubic')
+
+        # value and gradient work as expected
+        result1 = interp(x)
+        gradient1 = interp.gradient(x)
+        result_actual_1 = 0.2630309995970872
+        result_gradient_1 = np.array([0.22505535, -0.46465198, 0.02523666])
```

Comment:
What are these values and why are they correct? Maybe this test is trying to do two things at once: check against an independently calculated values (this is good! only please add at least a comment on how to reproduce these values), and somehow validate a random input. I'd say these should be two separate tests.